RELEASES.md

```diff
@@ -1,3 +1,128 @@
+Version 1.48.0 (2020-11-19)
+==========================
+
+Language
+--------
+
+- [The `unsafe` keyword is now syntactically permitted on modules.][75857] This
+  is still rejected *semantically*, but can now be parsed by procedural macros.
+
+Compiler
+--------
+- [Stabilised the `-C link-self-contained=<yes|no>`][76158] Which tells `rustc` whether to link
+  its own C runtime and libraries or to rely on a external linker to find them. (supported only on
+  `windows-gnu`, `linux-musl`, and `wasi` platforms.)
+- [You can now use `-C target-feature=+crt-static` on `linux-gnu` targets.][77386]
+- [Added tier 2\* support for `aarch64-unknown-linux-musl`.][76420]
+
+\* Refer to Rust's [platform support page][forge-platform-support] for more
+information on Rust's tiered platform support.
+
+Libraries
+---------
+- [`io::Write` is now implemented for `&ChildStdin` `&Sink`, `&Stdout`,
+  and `&Stderr`.][76275]
+- [All arrays now implement `TryFrom<Vec<T>>`.][76310]
+- [The `matches!` macro now supports having a trailing comma.][74880]
+- [`Vec<A>` now implements `PartialEq<[B]>` where `A: PartialEq<B>`.][74194]
+- [Nearly all of `Cell`'s panicking functions now use the `#[track_caller]`
+  attribute.][77055]
+
+Stabilized APIs
+---------------
+- [`slice::as_ptr_range`]
+- [`slice::as_mut_ptr_range`]
+- [`VecDeque::make_contiguous`]
+- [`future::pending`]
+- [`future::ready`]
+
+The following previously stable methods are now `const fn`'s:
+
+- [`Option::is_some`]
+- [`Option::is_none`]
+- [`Option::as_ref`]
+- [`Result::is_ok`]
+- [`Result::is_err`]
+- [`Result::as_ref`]
+- [`Ordering::reverse`]
+- [`Ordering::then`]
+
+Cargo
+-----
+
+Misc
+----
+- [You can now link to different items in `rustdoc` using the intra-doc link
+  syntax.][74430] E.g. ``/// Uses [`std::future`]`` will automatically generate
+  a link to `std::future`'s documentation. See ["Linking to items by
+  name"][intradoc-links] for more information.
+- [You can now specify `#[doc(alias = "<alias>")]` on items to add search aliases
+  when searching through `rustdoc`'s UI.][75740]
+- [You can now use `rustup install <major>.<minor>` to specify installing the
+  latest available patch of that minor version of the toolchain.][76107] E.g.
+  `rustup install 1.45` would install `1.45.2`, and `1.46` would install `1.46.0`.
+
+Compatibility Notes
+-------------------
+- [`const fn`s are now implicitly promoted to `const`.][75502] Meaning that it
+  will only warn if your code fails `const` evaluation, and not produce an error.
```

Comment:
I think this isn't quite right -- what has changed is that const fn will no longer get implicitly promoted, I think? i.e., `&foo()` where foo is a const fn and the type is requested at `&'static T` will no longer compile on beta.

cc @rust-lang/wg-const-eval is there a succinct summary we could include here? Maybe we can get someone to put together a short inside rust blog post we can link to here?

Reply:
Yeah that description does not seem to have much to do with what the PR does.

A short summary could be something like: promotion of references to `'static` lifetime inside `const fn` now follows the same rules as inside `fn`; in particular, `&foo()` will not be promoted to `'static` lifetime any more inside `const fn`.

This is a bugfix, we should never have done this to begin with. It is part of an ongoing effort to fix promotion, and then to restrict it a bit further to make it less painful throughout the compiler (see https://github.com/rust-lang/const-eval/issues/53).